packages/core/src/external/hl7-notification/datetime.ts

```diff
@@ -0,0 +1,151 @@
+import { Hl7Field, Hl7Message, Hl7Segment } from "@medplum/core";
+import { buildDayjsTz } from "@metriport/shared/common/date";
+import { errorToString } from "@metriport/shared";
+import { flow } from "lodash";
+import { getSendingApplication } from "../../command/hl7v2-subscriptions/hl7v2-to-fhir-conversion/msh";
+import { capture, out } from "../../util";
+import { getHieTimezone } from "./hie-timezone";
+
+const MSH_DATETIME_OF_MESSAGE_INDEX = 7;
+const EVN_RECORDED_DATETIME_INDEX = 2;
+const EVN_DATETIME_PLANNED_EVENT_INDEX = 3;
+const EVN_EVENT_OCCURRED_INDEX = 6;
+const PV1_ADMIT_DATE_INDEX = 44;
+const PV1_DISCHARGE_DATE_INDEX = 45;
+const PID_DEATH_DATETIME_INDEX = 29;
+const NK1_ROLE_START_DATE_INDEX = 8;
+const NK1_ROLE_END_DATE_INDEX = 9;
+const PV2_EXPECTED_ADMIT_DATE_INDEX = 8;
+const PV2_EXPECTED_DISCHARGE_DATE_INDEX = 9;
+const DG1_DIAGNOSIS_DATETIME_INDEX = 5;
+
+export function utcifyHl7Message(message: Hl7Message): Hl7Message {
+  const sendingApplication = getSendingApplication(message);
+  const hiePartner = sendingApplication ?? "Unknown HIE";
+  capture.setExtra({ sendingApplication });
+
+  /**
+   * TODO: Properly timezoneify the message using getHieTimezone, which requires making response format requests over email
+   * Temporary to get NewYorkHie ADTs correct for now.
+   * */
+  const resolvedTimezone = hiePartner !== "HIXNY" ? getHieTimezone(hiePartner) : "America/New_York";
```

Comment:
let's use boolean in the truthly format when possible - in this case, if partner is HXNY do this, otherwise that

or, even better, build a map w/ the parner and the tz :)

Reply:
BTW, can use offset instead of named TZs? Simpler, doesn't require people to think, remember, google it

Reply:
yep IANA timezones (America/New_York) adjust based on daylight savings